pkgs/applications/audio/stargate/default.nix

```
@@ -0,0 +1,169 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, fetchpatch
+, substituteAll
+, alsa-lib
+, autoconf
+, automake
+, ffmpegSupport ? true
+, ffmpeg
+, fftw
+, fftwSinglePrec
+, file
+, gdb
+, gettext
+, git
+, jq
+, lameSupport ? true
+, lame
+, libsbsms
+, libsndfile
+, pkg-config
+, portaudio
+, portmidi
+, python3
+, python3Packages
+, rubberband
+, stargate-libcds
+, vorbisToolsSupport ? true
+, vorbis-tools
+}:
+
+assert ffmpegSupport -> ffmpeg != null;
+assert lameSupport -> lame != null;
+assert vorbisToolsSupport -> vorbis-tools != null;
+
+let
+  python3CommonInputs = python3.withPackages
+    (ps: with ps; [
+      jinja2
+      mido
+      mutagen
+      numpy
+      psutil
+      pymarshal
+      pyqt5
+      pyyaml
+      wavefile
+      wheel
+    ]);
+in
+stdenv.mkDerivation rec {
+  pname = "stargate";
+  version = "21.11.5";
+  disabled = python3Packages.pythonOlder "3.6";
+
+  src = fetchFromGitHub {
+    owner = "stargateaudio";
+    repo = pname;
+    rev = "release-${version}";
+    sha256 = "sha256-d5ZuDCyq9gvfZvq57XHKEq0pvi+dWgz6z34kFIOmj7A=";
+  };
+
+  nativeBuildInputs = [
+    autoconf
+    automake
+    gdb
+    git
+    jq
+    pkg-config
+    python3Packages.cython
+    python3Packages.setuptools
+  ];
+
+  buildInputs = [
+    fftw
+    fftwSinglePrec
+    gettext
+    libsbsms
+    libsndfile
+    portaudio
+    portmidi
+    python3
+    rubberband
+    stargate-libcds
+    file
+  ]
+  ++ lib.lists.optional stdenv.isLinux alsa-lib
+  ++ lib.lists.optional ffmpegSupport ffmpeg
+  ++ lib.lists.optional lameSupport lame
+  ++ lib.lists.optional vorbisToolsSupport vorbis-tools;
+
+  propagatedBuildInputs = [
+    python3CommonInputs
+  ];
+
+  checkInputs = [
+    python3CommonInputs
+    python3Packages.pytest
+    python3Packages.pytest-cov
+    python3Packages.yq
+  ];
+
+  patches = [
+    # Remove vendored python dependencies
+    # Remove unecessary tests (coverage, valgrind)
+    ./Makefile.patch
+
+    # Backport fix for incorrect import
+    (fetchpatch {
+      name = "test_ctypes.py.patch";
+      url = "https://github.com/stargateaudio/stargate/commit/1641e4047610c44d23bee40ed16ec5d3b42534dc.patch";
+      sha256 = "sha256-Be1D4cmWPDx7TX3M63vjAi872FF1r0J5jY3qZtZxM60=";
+    })
+
+    # Patch paths of shared libraries and binaries.
+    (substituteAll {
+      src = ./path.patch;
+      libportaudio =
+        "${lib.getLib portaudio}/lib/libportaudio${stdenv.hostPlatform.extensions.sharedLibrary}";
+
+      libportmidi =
+        "${lib.getLib portmidi}/lib/libportmidi${stdenv.hostPlatform.extensions.sharedLibrary}";
+
+      file = "${file}/bin/file";
+    })
+  ];
+
+  preConfigure = ''
+  cd src
+  '';
+
+  SRC_SG_PY_VENDOR_DIR = "sg_py_vendor";
+
+  doCheck = true;
+
+  preCheck = ''
+  # Fix Permission denied: /homeless-shelter
+  # https://github.com/NixOS/nixpkgs/issues/12591
+  mkdir --parents check-phase
+  export HOME=$(pwd)/check-phase
```

```suggestion
  mkdir -p check-phase
  export HOME=$PWD/check-phase
```
I would do it like this but not important as well.